sql/catalyst/src/test/scala/org/apache/spark/sql/util/TimestampFormatterSuite.scala

```diff
@@ -235,7 +237,7 @@ class TimestampFormatterSuite extends SparkFunSuite with SQLHelper with Matchers
   }
 
   test("SPARK-30958: parse timestamp with negative year") {
-    val formatter1 = TimestampFormatter("yyyy-MM-dd HH:mm:ss", ZoneOffset.UTC)
+    val formatter1 = TimestampFormatter("yyyy-MM-dd HH:mm:ss", ZoneOffset.UTC, true)
```

Comment:
is the point to test the actual parser? with `needVarLengthSecondFraction = true`.

Reply:
yes